apps/ledger-live-desktop/src/renderer/components/Web3AppWebview/helpers.ts

```diff
@@ -0,0 +1,225 @@
+
+  const headers = useMemo(() => {
+    return getClientHeaders({
+      client: "ledger-live-desktop",
+      theme,
+    });
+  }, [theme]);
+  */
+
+  useImperativeHandle(
+    webviewAPIRef,
+    () => {
+      return {
+        reload: () => {
+          const webview = safeGetRefValue(webviewRef);
+
+          webview.reload();
+        },
+        goBack: () => {
+          const webview = safeGetRefValue(webviewRef);
+
+          webview.goBack();
+        },
+        goForward: () => {
+          const webview = safeGetRefValue(webviewRef);
+
+          webview.goForward();
+        },
+        openDevTools: () => {
+          const webview = safeGetRefValue(webviewRef);
+
+          webview.openDevTools();
+        },
+        loadURL: (url: string): Promise<void> => {
+          const webview = safeGetRefValue(webviewRef);
+
+          return webview.loadURL(url);
+        },
+        clearHistory: () => {
+          const webview = safeGetRefValue(webviewRef);
+
+          webview.clearHistory();
+        },
+      };
+    },
+    [webviewRef],
+  );
+
+  const [isMounted, setMounted] = useState<boolean>(false);
+  useEffect(() => {
+    setMounted(true);
+  }, [isMounted]);
+
+  const handlePageTitleUpdated = useCallback((event: Electron.PageTitleUpdatedEvent) => {
+    setState(oldState => ({
+      ...oldState,
+      title: event.title,
+    }));
+  }, []);
+
+  const handleDidNavigateInPage = useCallback(
+    (event: Electron.DidNavigateInPageEvent) => {
+      const webview = webviewRef.current;
+
+      if (!webview) {
+        return;
+      }
+
+      setState(oldState => ({
+        ...oldState,
+        url: event.url,
+        canGoBack: webview.canGoBack(),
+        canGoForward: webview.canGoForward(),
+      }));
+    },
+    [webviewRef],
+  );
+
+  const handleDidNavigate = useCallback(
+    (event: Electron.DidNavigateEvent) => {
+      const webview = webviewRef.current;
+
+      if (!webview) {
+        return;
+      }
+
+      setState(oldState => ({
+        ...oldState,
+        url: event.url,
+        canGoBack: webview.canGoBack(),
+        canGoForward: webview.canGoForward(),
+      }));
+    },
+    [webviewRef],
+  );
+
+  const handleDidStartLoading = useCallback(() => {
+    setState(oldState => ({
+      ...oldState,
+      loading: true,
+    }));
+  }, []);
+
+  const handleDidStopLoading = useCallback(() => {
+    setState(oldState => ({
+      ...oldState,
+      loading: false,
+    }));
+  }, []);
+
+  const handleDomReady = useCallback(() => {
+    const webview = webviewRef.current;
+
+    if (!webview) {
+      return;
+    }
+
+    setState({
+      url: webview.getURL(),
+      canGoBack: webview.canGoBack(),
+      canGoForward: webview.canGoForward(),
+      title: webview.getTitle(),
+      loading: webview.isLoading(),
+    });
+  }, [webviewRef]);
+
+  useEffect(() => {
+    const webview = webviewRef.current;
+
+    if (!isMounted || !webview) {
+      return;
+    }
```

Comment:
Not a big fan of this `isMounted` solution to wait and do a second render.
I suppose we can start with that and look for something better later.

Reply:
Agreed